src/main/cpp/src/hive_hash.cu

```diff
@@ -469,6 +519,40 @@ void check_nested_depth(cudf::table_view const& input)
   }
 }
 
+void flatten_table(std::vector<col_info>& col_infos,
```

Comment:
Combine the `check_nested_depth` and `flatten_table`,  we can check depth when flatten table.